### PR TITLE
Reports octane another

### PR DESCRIPTION
**Pull request by kevinhinterlong**
## Description
Octane common-actions in reports

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/reports/addon/components/common-actions/schedule.js [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
- *    onSave=(action 'onSave')
- *    onRevert=(action 'onRevert')
- *    onDelete=(action 'onDelete')
- * }}
+ * <CommonActions::Schedule
+ *    @model={{this.model}}
+ *    @onSave={{this.onSave}}
+ *    @onRevert={{this.onRevert}}
+ *    @onDelete={{this.onDelete}}
+ * >
```
Comment (daniellevinson) [inline packages/reports/addon/components/common-actions/schedule.js:10]:
```suggestion
 * />
```

File: packages/reports/addon/components/common-actions/schedule.js [hunk 2]
```diff
@@ -17,54 +17,57 @@ import layout from '../../templates/components/common-actions/schedule';
...
-  layout,
-
+@templateLayout(layout)
+@tagName('')
+class ScheduleActionComponent extends Component {
   /**
    * @property {Service} store
    */
-  store: service(),
+  @service() store;
```
Comment (daniellevinson) [inline packages/reports/addon/components/common-actions/schedule.js:34]:
```suggestion
  @service store;
```

File: packages/reports/addon/components/common-actions/schedule.js [hunk 3]
```diff
@@ -17,54 +17,57 @@ import layout from '../../templates/components/common-actions/schedule';
...
 
   /**
    * @property {Service} user
    */
-  user: service(),
+  @service() user;
```
Comment (daniellevinson) [inline packages/reports/addon/components/common-actions/schedule.js:39]:
```suggestion
  @service user;
```

File: packages/reports/addon/components/common-actions/schedule.js [hunk 4]
```diff
@@ -17,54 +17,57 @@ import layout from '../../templates/components/common-actions/schedule';
...
 
   /**
    * @property {Service} naviNotifications
    */
-  naviNotifications: service(),
+  @service() naviNotifications;
```
Comment (daniellevinson) [inline packages/reports/addon/components/common-actions/schedule.js:44]:
```suggestion
  @service naviNotifications;
```

File: packages/reports/addon/components/common-actions/share.js [hunk 5]
```diff
@@ -1,45 +1,48 @@
...
  *
  * Usage:
- *   {{#common-actions/share
- *      pageTitle=nameOfPageToShare
- *      buildUrl=(action 'optionalCodeForGeneratingUrlToShare')
- *   }}
+ *   <CommonActions::Share
+ *     @pageTitle={{this.nameOfPageToShare}}
+ *     @buildUrl={{this.optionalCodeForGeneratingUrlToShare}}
+ *   >
```
Comment (daniellevinson) [inline packages/reports/addon/components/common-actions/share.js:9]:
```suggestion
 *   />
```

File: packages/reports/addon/templates/components/common-actions/delete.hbs [hunk 6]
```diff
@@ -1,30 +1,30 @@
...
+    <div class="btn-container">
+      <button
+        type="button"
+        class="btn btn-primary"
+        disabled={{this.isDeleting}}
+        {{on "click" (pipe (toggle "isDeleting" this) (fn @deleteAction @model) this.closeModal)}}
+      >
+        Confirm
+      </button>
+      <button type="button" class="btn btn-secondary" onclick={{toggle "showModal" this}} disabled={{this.isDeleting}}>Cancel</button>
```
Comment (daniellevinson) [inline packages/reports/addon/templates/components/common-actions/delete.hbs:27]:
```suggestion
      <button type="button" class="btn btn-secondary" {{on "click" (toggle "showModal" this)}} disabled={{this.isDeleting}}>Cancel</button>
```

File: packages/reports/addon/templates/components/common-actions/get-api.hbs [hunk 7]
```diff
@@ -1,53 +1,54 @@
 {{!-- Copyright 2020, Yahoo Holdings Inc. Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms. --}}
+<div class="get-api" ...attributes>
+  {{!-- Check wether we need to disable the button click --}}
+  <button class="get-api__btn {{@buttonClassNames}}" type="button" disabled={{@disabled}}
+    onclick={{queue (optional @beforeAction) (toggle "showModal" this)}}>
```
Comment (daniellevinson) [inline packages/reports/addon/templates/components/common-actions/get-api.hbs:5]:
```suggestion
    {{on "click" (queue (optional @beforeAction) (toggle "showModal" this))}}>
```

**Comment by cythrawll** [comment]
Approved after Daniel's comments are addressed.

**Pull request merged**